pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,72 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def _complete_unitary(columns):
+    """
+    Completes a unitary matrix given a list of orthonormal columns.
+
+    Args:
+        columns (List[Array]): List of initial orthonormal columns of dimension d.
+
+    Returns:
+        Array: Completed unitary matrix of dimension :math:`(d, d)`.
+    """
+
+    columns = qml.math.stack(columns).T
+    d = columns.shape[0]
+    k = columns.shape[1]
+
+    unitary = qml.math.zeros_like(columns @ columns.T)
+
+    if qml.math.get_interface(columns) == "jax":
+        unitary = unitary.at[:, :k].set(columns)
+
+    else:
+        unitary[:, :k] = columns
+
+    # Complete the remaining columns using Gram-Schmidt
+    rng = np.random.default_rng(42)
```

Comment:
Does this need to be seeded? Won't this ensure that each and every call to this will use the same correlated RNG outputs for the same input? 
If `None` it will pull entropy from the operating system's default source, which is likely a better source of randomness. Or, allowing the provision of a seed as a function input would be the best option, as this ensure control is at the caller's level. What do you think?

Reply:
I had to do it or `qml.assert_valid` complained that the decomposition did not match the operator in the queue (two different Unitaries were generated)

Reply:
Assert valid in the tests? But we should be able to specify the seed to ensure the tests generate the same output? I'd expect numerical differences in the outcomes with generators that are seeded differently.

Reply:
Actually these values of the matrices are not used for anything, we generate them just to be able to put the matrix in QubitUnitary. Since it is also a private function, I think it is ok to set it in this case :) User does not need to know about it